packages/node/src/lock/lock.ts

```diff
@@ -0,0 +1,49 @@
+import Queue from "p-queue";
+import uuid from "uuid";
+
+import { Deferred } from "../deferred";
+
+export class Lock {
```

Comment:
NOTE: This is just a helper class for the `MemoryLockService` example.

Reply:
Makes sense to have an in-memory implementation for unit tests.

Reply:
I still need to review the `Lock` object. It seems unnecessary to me conceptually but I still need to dig into that class.